src/cmdstan/command.hpp

```diff
@@ -155,7 +158,57 @@ namespace cmdstan {
     std::shared_ptr<stan::io::var_context> init_context = get_var_context(init);
 
     int return_code = stan::services::error_codes::CONFIG;
-    if (parser.arg("method")->arg("diagnose")) {
+
+    if (parser.arg("method")->arg("generate_quantities")) {
+      // read sample from cmdstan csv output file
+      string_argument* fitted_params_file =
+        dynamic_cast<string_argument*>(parser.arg("method")->arg("generate_quantities")->arg("fitted_params"));
+      if (fitted_params_file->is_default()) {
+        info("Must specify argument fitted_params which is a csv file containing the sample.");
+        return_code = stan::services::error_codes::CONFIG;
+      }
+      std::string fname(fitted_params_file->value());
+      std::fstream stream(fname.c_str(), std::fstream::in);
+      std::stringstream msg;
+      if (fname != "" && (stream.rdstate() & std::ifstream::failbit)) {
+        msg << "Can't open specified file, \"" << fname << "\"" << std::endl;
+        throw std::invalid_argument(msg.str());
+      }
+      stan::io::stan_csv fitted_params;
+      stan::io::stan_csv_reader::read_metadata(stream, fitted_params.metadata, &msg);
+      if (!stan::io::stan_csv_reader::read_header(stream, fitted_params.header, &msg)) {
+        msg << "Error reading fitted param names from sample csv file \"" << fname << "\"" << std::endl;
+        throw std::invalid_argument(msg.str());
+      }
+      stan::io::stan_csv_reader::read_adaptation(stream, fitted_params.adaptation, &msg);
+      fitted_params.timing.warmup = 0;
+      fitted_params.timing.sampling = 0;
+      stan::io::stan_csv_reader::read_samples(stream, fitted_params.samples, fitted_params.timing, &msg);
+      stream.close();
+
+      std::vector<std::string> param_names;
+      model.constrained_param_names(param_names, false, false);
+      size_t num_cols = param_names.size();
+      size_t num_rows = fitted_params.metadata.num_samples;
+
+      // check that all parameter names are in sample, in order 
+      if (num_cols + 7 > fitted_params.header.size()) {
+        msg << "Mismatch between model and fitted_parameters csv file \"" << fname << "\"" << std::endl;
+        throw std::invalid_argument(msg.str());
+      }
+      for (size_t i = 0; i < num_cols; ++i) {
+        if (param_names[i].compare(fitted_params.header[i + 7]) != 0) {
+          msg << "Mismatch between model and fitted_parameters csv file \"" << fname << "\"" << std::endl;
+          throw std::invalid_argument(msg.str());
+        }
+      }
+      return_code = stan::services::standalone_generate(model,
```

Comment:
[optional]
If you're in the `stan` namespace, you can drop the `stan::` prefix.  Your call for clarity.

Reply:
leaving as is - consistent with rest of calls to service methods in this file.